src/Features/Core/Portable/Snippets/ISnippetProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    internal interface ISnippetProvider
+    {
+        /// Determines if a snippet can exist at a particular location.
+        Task<SnippetData?> GetSnippetDataAsync(Document document, int position, CancellationToken cancellationToken);
```

Comment:
sounds like this can just return a boolean then, right?

Reply:
I use it to create a list of available snippets for a particular location, so I need the information stored in the data type.